src/storage/store/column.cpp

```diff
@@ -520,21 +500,20 @@ PageCursor Column::getPageCursorForOffsetInGroup(offset_t offsetInChunk, const R
     return pageCursor;
 }
 
-WALPageIdxPosInPageAndFrame Column::createWALVersionOfPageForValue(
-    node_group_idx_t nodeGroupIdx, offset_t offsetInChunk) {
+void Column::updatePageWithOffset(node_group_idx_t nodeGroupIdx, offset_t offsetInChunk,
+    const std::function<void(uint8_t*, offset_t)>& writeOp) {
     auto originalPageCursor =
         getPageCursorForOffset(TransactionType::WRITE, nodeGroupIdx, offsetInChunk);
     bool insertingNewPage = false;
     if (originalPageCursor.pageIdx == INVALID_PAGE_IDX) {
-        return {{INVALID_PAGE_IDX, INVALID_PAGE_IDX, nullptr}, originalPageCursor.elemPosInPage};
+        return writeOp(nullptr, originalPageCursor.elemPosInPage);
     } else if (originalPageCursor.pageIdx >= dataFH->getNumPages()) {
         KU_ASSERT(originalPageCursor.pageIdx == dataFH->getNumPages());
         DBFileUtils::insertNewPage(*dataFH, dbFileID, *bufferManager, *wal);
         insertingNewPage = true;
     }
-    auto walPageIdxAndFrame = DBFileUtils::createWALVersionIfNecessaryAndPinPage(
-        originalPageCursor.pageIdx, insertingNewPage, *dataFH, dbFileID, *bufferManager, *wal);
-    return {walPageIdxAndFrame, originalPageCursor.elemPosInPage};
+    DBFileUtils::updatePage(*dataFH, dbFileID, originalPageCursor.pageIdx, insertingNewPage,
+        *bufferManager, *wal, [&](auto frame) { writeOp(frame, offsetInChunk); });
```

Comment:
Issue was passing the offsetInChunk instead of the `originalPageCursor.elemPosInPage`.